knowledge-base/grid-search-column-menu.md

```diff
@@ -0,0 +1,168 @@
+---
+title: Change Grid Search Results on Column Hide or Show
+description: Learn how to modify the Grid search results on the fly when the user hides or shows a column.
+type: how-to
+page_title: How to Change Grid Search Results on Column Hide or Show
+slug: grid-kb-search-match-visible-columns
+tags: telerik, blazor, grid, search
+ticketid: 1565592
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor, <br /> TreeList for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB article answers the following questions:
+
+* How to change the Grid search results when the user hides or shows a string column?
+* How to refresh and match the Grid search results when the `Visible` attribute of a column changes?
+* How to hide or show search results in the Grid when the visibility of a column changes and a column is no longer displayed?
+
+## Solution
+
+1. Subscribe to the [Grid `OnStateChanged` event]({%slug grid-state%}#onstatechanged).
+1. Check if the `PropertyName` event argument is equal to `"ColumnStates"` to verify that the user modified the column state.
```

Comment:
I know that our style guide advocates for using simple tenses. However, in this case the perfect tense fits better grammatically.

```suggestion
1. Check if the `PropertyName` event argument is equal to `"ColumnStates"` to verify that the user has modified the column state.
```